src/Microsoft.PowerShell.Commands.Utility/commands/utility/XmlCommands.cs

```diff
@@ -359,6 +362,246 @@ protected override void StopProcessing()
+            }
+        }
+
+        #endregion stream
+
+        #region IDisposable Members
+
+        /// <summary>
+        /// Set to true when object is disposed.
+        /// </summary>
+        private bool _disposed;
+
+        /// <summary>
+        /// Public dispose method.
+        /// </summary>
+        public void Dispose()
+        {
+            if (_disposed == false)
+            {
+                CleanUp();
+            }
+
+            _disposed = true;
+        }
+
+        #endregion IDisposable Members
+    }
+    #endregion ConvertTo-Clixml
+
+    #region ConvertFrom-Clixml
+
+    /// <summary>
+    /// Implements ConvertFrom-Clixml command.
+    /// </summary>
+    [Cmdlet(VerbsData.ConvertFrom, "Clixml", SupportsPaging = true, HelpUri = "")]
+    public sealed class ConvertFromClicmlCommand : PSCmdlet, IDisposable
+    {
+        #region Command Line Parameters
+
+        /// <summary>
+        /// Gets or sets input object which is written Clixml format.
+        /// </summary>
+        [Parameter(Position = 0, Mandatory = true, ValueFromPipeline = true, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public PSObject[] InputObject { get; set; }
+
+        #endregion Command Line Parameters
+
+        private ImportXmlHelper _importXmlHelper;
+        private StringBuilder _stringBuilder;
+
+        #region Overrides
+
+        /// <summary>
+        /// Begin processing.
+        /// </summary>
+        protected override void BeginProcessing()
+        {
+            _stringBuilder = new StringBuilder();
+        }
+
+        /// <summary>
+        /// Process record overload.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            string temp = InputObject[0].ToString();
+            _stringBuilder.AppendLine(temp);
+        }
+
+        /// <summary>
+        /// End processing.
+        /// </summary>
+        protected override void EndProcessing()
+        {
+            _importXmlHelper = new ImportXmlHelper(this, _stringBuilder.ToString());
+            _importXmlHelper.Import();
+        }
+
+        /// <summary>
+        /// Stop processing.
+        /// </summary>
+        protected override void StopProcessing()
+        {
+            base.StopProcessing();
+            _importXmlHelper.Stop();
+        }
+
+        #endregion Overrides
+
+        #region IDisposable Members
+
+        /// <summary>
+        /// Set to true when object is disposed.
+        /// </summary>
+        private bool _disposed;
+
+        /// <summary>
+        /// Public dispose method.
+        /// </summary>
+        public void Dispose()
+        {
+            if (_disposed == false)
+            {
+                if (_importXmlHelper != null)
+                {
+                    _importXmlHelper.Dispose();
+                    _importXmlHelper = null;
+                }
+            }
+
+            _disposed = true;
+        }
```

Comment:
It's unusual for a `Dispose()` method to throw, but iirc it can happen; I'd prefer to see this with a try/finally so that `_disposed` is always set correctly.

Reply:
It seems `_importXmlHelper.Dispose()` can not throw.